krux_cloud_formation/cloud_formation.py

```diff
@@ -200,10 +200,11 @@ def save(self, stack_name, s3_key=None):
         :param s3_key: :py:class:`str` Name of the s3 file to be used to upload the template. If set to None, stack_name is used.
         """
         key = s3_key or stack_name
-        s3_file = self._s3.create_key(bucket_name=self._bucket_name, key=key, str_content=self.template.to_json())
 
         if self._is_stack_exists(stack_name):
             try:
```

Comment:
When there is no change to the template file, the cloud formation throws a specific exception rather than handling the noop gracefully. The `try` and `catch` are here to catch that and return.